lazyflow/utility/io_util/RESTfulPrecomputedChunkedVolume.py

```diff
@@ -109,58 +103,34 @@ def _init_config(self, volume_url=None, volume_description=None):
             self._json_info = volume_description
 
         jsonschema.validate(self._json_info, self.info_schema)
-        _scale_info = {x["key"]: x for x in self._json_info["scales"]}
-
-        # save json contents
-        self._scale_info = _scale_info
-
-        self._use_scale, resolution = self.determine_lowest_scale(_scale_info)
-
-        self.available_scales = self._scale_info.keys()
 
+        self.scales = self._json_info["scales"]
+        # Sort from lowest to highest resolution along x
+        # scales["resolution"] is in nanometers-per-pixel. So bigger = lower resolution, hence reverse sort.
+        self.scales.sort(key=lambda s: s["resolution"][0], reverse=True)
```

Comment:
from [the precomputed specs](https://github.com/google/neuroglancer/blob/master/src/neuroglancer/datasource/precomputed/volume.md#info-json-file-specification):
> "resolution": 3-element array [x, y, z] of numeric values specifying the x, y, and z dimensions of a voxel in nanometers. The x, y, and z "resolution" values must not decrease as the index into the "scales" array increases.

I don't think you should sort anything here, as the order there should be sorted already. Also sorting only by the first element could not work (values are only guaranteed not to decrease, but `x` does not necessarily increase in all datasets - according to the definition).

I see that it was also in the previous code, but - I don't think it should be necessary.

Reply:
We do need to reverse it, because the spec requires increasing "resolution" values, which means the first scale in the array is the original scale. We want to start with the opposite, lowest (most downscaled) and list the scales increasing from there. But agree that we should reverse whatever order the server gives us considering it's required to be sorted in some meaningful way.